backends/ebpf/psa/ebpfPsaTable.cpp

```diff
@@ -293,51 +293,35 @@ ActionTranslationVisitor* EBPFTablePSA::createActionTranslationVisitor(
 
 void EBPFTablePSA::emitValueStructStructure(CodeBuilder* builder) {
     if (implementation != nullptr) {
-        // TODO: add priority for ternary table
-
+        if (isTernaryTable()) {
+            builder->emitIndent();
+            builder->append("__u32 priority;");
+            builder->newline();
+        }
         implementation->emitReferenceEntry(builder);
     } else {
         EBPFTable::emitValueStructStructure(builder);
     }
 }
 
 void EBPFTablePSA::emitInstance(CodeBuilder *builder) {
-    if (keyGenerator != nullptr) {
+    if (isTernaryTable()) {
+        emitTernaryInstance(builder);
+    } else {
         TableKind kind = isLPMTable() ? TableLPMTrie : TableHash;
-        emitTableDecl(builder, instanceName, kind,
+        builder->target->emitTableDecl(builder, instanceName, kind,
```

Comment:
What about spinlock for meters?

Reply:
Can you clarify?

Reply:
I see that tests are passing, but I don't find a place where spinlocks are emitted for DirectMeters.